examples/tensor/fixed_rank_prod_expressions.cpp

```diff
@@ -0,0 +1,169 @@
+//
+//  Copyright (c) 2018-2020, Cem Bassoy, cem.bassoy@gmail.com
+//  Copyright (c) 2019-2020, Amit Singh, amitsingh19975@gmail.com
+//
+//  Distributed under the Boost Software License, Version 1.0. (See
+//  accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+//  The authors gratefully acknowledge the support of
+//  Google
+//
+
+
+#include <boost/numeric/ublas/tensor.hpp>
+#include <boost/numeric/ublas/matrix.hpp>
+#include <boost/numeric/ublas/vector.hpp>
+#include <iostream>
+
+int main()
+{
+    using namespace boost::numeric::ublas;
+
+    using format_t  = column_major;
+    using value_t   = float; // std::complex<double>;
+    using matrix_t = matrix<value_t,format_t>;
+    using vector_t = vector<value_t>;
+
+    // Tensor-Vector-Multiplications - Including Transposition
+    // dynamic_extents with static rank
+    {
+
+        auto n = dynamic_extents<3>{3,4,2};
+        auto A = fixed_rank_tensor(n,value_t(2));
+        using tensor_t = fixed_rank_tensor< value_t, 2>;
+        auto q = 0u; // contraction mode
+
+        // C1(j,k) = T2(j,k) + A(i,j,k)*T1(i);
+        q = 1u;
+        tensor_t C1 = matrix_t(n[1],n[2],2) + prod(A,vector_t(n[q-1],1),q);
```

Comment:
Can we use `auto` instead of `tensor_t` and get rid of the `using` expression in line 34?

Reply:
this returns a tensor_expression which I don't think without we can use `std::cout`.

Reply:
can we have `operator<<` overloading for printing `tensor_expression`?

Reply:
> this returns a tensor_expression which I don't think without casting it back to tensor we can use `std::cout`.

Yes, I forgot. I am okay with that then.

> can we have `operator<<` overloading for printing `tensor_expression`?

Not sure if we its overloaded for `tensor_expression`. Lets mark it as an enhancment issue for later.

Reply:
I think Ashar's expression template will enhance it.